integration/bip0009_test.go

```diff
@@ -268,7 +272,33 @@ func testBIP0009(t *testing.T, forkKey string, deploymentID uint32) {
 	if err != nil {
 		t.Fatalf("failed to generated block: %v", err)
 	}
-	assertChainHeight(r, t, (confirmationWindow*4)-1)
+	expectedChainHeight := (confirmationWindow * 4) - 1
+	assertChainHeight(r, t, expectedChainHeight)
+
+	// If this isn't a fork that has a min activation height set, then it
+	// should be active at this point.
+	if deployment.MinActivationHeight == 0 {
+		assertSoftForkStatus(r, t, forkKey, blockchain.ThresholdActive)
+		return
+	}
+
+	// Otherwise, we'll need to mine additional blocks to pass the min
+	// activation height and ensure the rule set applies. For regtest the
+	// deployment can only activate after height 600, and at this point
+	// we've mined 4*144 blocks, so another confirmation window will put us
+	// over.
+	numBlocksLeft := confirmationWindow
+	for i := uint32(0); i < numBlocksLeft; i++ {
+		_, err := r.GenerateAndSubmitBlock(nil, signalForkVersion,
+			time.Time{})
+		if err != nil {
+			t.Fatalf("failed to generated block %d: %v", i, err)
+		}
```

Comment:
Before generating and submitting a block, should we assert that we're still in state `blockchain.ThresholdLockedIn`? This should be true until after we mine the very last block (after which we exit the loop).

Reply:
Good idea, added on.